File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    let index1 = 0;  // Pointer for cards1
    let index2 = 0;  // Pointer for cards2
    
    for (let i = 0; i < goal.length; i++) {
        if (index1 < cards1.length && goal[i] === cards1[index1]) {
            index1++;
        } else if (index2 < cards2.length && goal[i] === cards2[index2]) {
            index2++;
        } else {
            return 'No';
        }
    }
    return 'Yes';
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
from collections import deque
def solution(cards1, cards2, goal):
    cards1 = deque(cards1)
    cards2 = deque(cards2)
    goal = deque(goal)
    answer = "Yes"

    for i in range(len(goal)):
        if cards1 and goal[0] == cards1[0]:
            goal.popleft()
            cards1.popleft()
        elif cards2 and goal[0] == cards2[0]:
            goal.popleft()
            cards2.popleft()
        else:
            answer = "No"
    return answer